source/js/components/petition/donation-modal.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import classNames from 'classnames';
+
+class DonationModal extends React.Component {
+
+  render() {
+    return (
+      <div className="modal-underlay">
+        <div className="modal show" role="dialog">
+          <div className="modal-dialog modal-lg" role="document">
+            {this.getModalContent()}
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  getModalContent() {
+    if (!this.donateURL) {
+      let query = [
+        `ctn=${this.props.ctn}`,
+        `dmi=${this.props.dmi}`
+      ].join(`&`);
+      this.donateURL = `https://donate.mozilla.org/en-US/?${query}`;
+    }
+    return (
+      <div className="modal-content" tabIndex="0">
+        <div className="modal-header text-right">
+          <button className="close" data-dismiss="modal" aria-label="Close" onClick={this.props.onClose}>
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+
+        <div className="modal-body">
+          <h3 className={ classNames(`h3-heading`, `text-center`) }>
+            { this.props.heading }
+          </h3>
+          <p className={ classNames(`body-large`, `text-center`) }>
+            {this.props.bodyText}
+          </p>
+        </div>
+
+        <div className="text-center">
+          <a className="btn btn-normal" href={this.donateURL} target="_blank" onClick={e => this.userElectedToDonate(e)}>
```

Comment:
I've made this an `<a>` because it means people get to see where this will lead them with a mouseover, but we can make this a `<button>`, too. As an actual navigating element, though, `<a>` feels way more semantically correct.